static/js/stream_data.js

```diff
@@ -523,15 +531,23 @@ export function id_is_subscribed(stream_id) {
 export function get_stream_privacy_policy(stream_id) {
     const sub = sub_store.get(stream_id);
 
-    if (!sub.invite_only) {
+    if (!sub.invite_only && !sub.is_web_public) {
```

Comment:
You can skip this change if you just move the `sub.is_web_public` check before this one.

Reply:
Done.